samples/cxx/custom/custom.cpp

```diff
@@ -0,0 +1,197 @@
+    }
+    
+    /**
+     * Number of equations in the ODE system.
+     *   - overrided from FuncEval, called by the integrator during initialization.
+     */
+    size_t neq() {
+        return nEqs;
+    }
+
+    /**
+     * Provide the current values of the state vector, *y*.
+     *   - overrided from FuncEval, called by the integrator during initialization.
+     * @param[out] y solution vector, length neq()
+     */
+    void getState(double* y) {
+        // the solution vector *y* is [T, Y_1, Y_2, ... Y_K], where T is the
+        // system temperature, and Y_k is the mass fraction of species k.
+        y[0] = gas->temperature();
+        gas->getMassFractions(&y[1]);
+    }
+
+private:
+    // private member variables, to be used internally.
+    shared_ptr<ThermoPhase> gas;
+    shared_ptr<Kinetics> kinetics;
+    vector_fp hbar;
+    vector_fp wdot;
+    double pressure;
+    int nSpecies;
+    int nEqs;
+};
+
+int main() {
+    /* -------------------- CREATE GAS & SPECIFY STATE -------------------- */
+    auto sol = newSolution("gri30.yaml");
+    auto gas = sol->thermo();
+    gas->setState_TPX(1001, OneAtm, "H2:2, O2:1, N2:4");
+
+    /* ---------------------- CREATE CSV OUTPUT FILE ---------------------- */
+    // simulation results will be outputted to a .csv file as complete state vectors
+    // for each simulation time point.
+    // create the csv file, overwriting any existing ones with the same name.
+    std::ofstream outputFile("custom_cxx.csv");
+    
+    // for convenience, a title for each of the state vector's components is written to
+    // the first line of the csv file.
+    outputFile << "time (s), temp (K), ";
+    for (int k = 0; k < gas->nSpecies(); k++)
+        outputFile << "Y_" << gas->speciesName(k) << ", ";
+    outputFile << std::endl;
+
+    /* --------------------- CREATE ODE RHS EVALUATOR --------------------- */
+    ReactorODEs odes = ReactorODEs(sol);
+
+    /* ---------------------- SPECIFY TIME INTERVAL ----------------------- */
+    // the simulation is run over the time interval specified below. tnow is initialized
+    // with the simulation's start time, and is updated on each timestep to reflect
+    // the new absolute time of the system.
+    double tnow = 0.0;
+    double tfinal = 1e-3;
+    
+    /* ------------------- CREATE & INIT ODE INTEGRATOR ------------------- */
+    // create an ODE integrator object, which will be used to solve the system of ODES defined
+    // in the ReactorODEs class. a C++ interface to the C-implemented SUNDIALS CVODES integrator
+    // (CVodesIntegrator) is built into Cantera, and will be used to solve this example.
+    //  - the default settings for CVodesIntegrator are used:
+    //     solution method: BDF_Method
+    //     problem type: DENSE + NOJAC
+    //     relative tolerance: 1.0e-9
+    //     absolute tolerance: 1.0e-15
+    //     max step size: +inf
+    Integrator *integrator = newIntegrator("CVODE");
```

Comment:
To avoid a memory leak here, write
```suggestion
    std::unique_ptr<Integrator> integrator(newIntegrator("CVODE"));
```